ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java

```diff
@@ -212,21 +194,52 @@ public APIGatewayProxyResponseEvent handleRequest(
                                     ipvAuthorisationService.validateResponse(
                                             input.getQueryStringParameters(),
                                             session.getSessionId()));
-            if (errorObject.isPresent()) {
-                return generateAuthenticationErrorResponse(
-                        authRequest,
-                        new ErrorObject(ACCESS_DENIED_CODE, errorObject.get().getDescription()),
-                        false,
-                        clientSessionId,
-                        session.getSessionId());
-            }
             var userProfile =
                     dynamoService
                             .getUserProfileFromEmail(session.getEmailAddress())
                             .orElseThrow(
                                     () ->
                                             new IpvCallbackException(
                                                     "Email from session does not have a user profile"));
+            var rpPairwiseSubject =
+                    segmentedFunctionCall(
+                            "calculatePairwiseSubject",
+                            () ->
+                                    ClientSubjectHelper.getSubject(
+                                            userProfile,
+                                            clientRegistry,
+                                            dynamoService,
+                                            configurationService.getInternalSectorUri()));
+
+            var internalPairwiseSubjectId =
+                    ClientSubjectHelper.calculatePairwiseIdentifier(
+                            userProfile.getSubjectID(),
+                            URI.create(configurationService.getInternalSectorUri()),
+                            dynamoService.getOrGenerateSalt(userProfile));
+
+            if (errorObject.isPresent()) {
+                var accountInterventionStatus =
+                        ipvCallbackHelper.getAccountInterventionStatus(internalPairwiseSubjectId);
+                if (configurationService.isAccountInterventionServiceEnabled()) {
+                    if (accountInterventionStatus.blocked()) {
+                        // TODO: back channel logout + redirect to blocked page
+                        LOG.info("Account is blocked");
+                    } else if (accountInterventionStatus.suspended()
+                            || accountInterventionStatus.resetPassword()
+                            || accountInterventionStatus.reproveIdentity()) {
+                        // TODO: back channel logout + redirect to suspended page
+                        LOG.info(
+                                "Account is suspended, requires a password reset, or requires identity to be reproved");
+                    }
+
+                    return ipvCallbackHelper.generateAuthenticationErrorResponse(
```

Comment:
This is currently only called if AIS is enabled - is that correct?

Reply:
Just saw that when addressing your comment above - it should be outside the conditional, shall move :)